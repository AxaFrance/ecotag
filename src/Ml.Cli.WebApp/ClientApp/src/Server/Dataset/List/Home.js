import React from 'react';
import { Link } from 'react-router-dom';
import Title from '../../../TitleBar';
import './Home.scss';
import ItemsTable from "./ItemsTable";
import EmptyArrayManager from "../../../EmptyArrayManager";

const Home = ({ items, filters, onChangePaging, onChangeFilter }) => {
    return (
        <>
            <Title title="Page datasets" subtitle="un dataset représente un essemble de données" />
            <div className="af-home container">
                <Link className="btn af-btn af-btn--quote" to="/datasets/new">
                    <span className="af-btn__text">Nouveau dataset</span>
                </Link>
                <h1 className="af-title--content">Vos datasets actifs : {items.length}</h1>
                <div className="row row--datasets-filters">
                    <div className="col">
                        <div className="af-filter-inline">
                            <span className="af-filter-inline__title">
                                <span className="glyphicon glyphicon-filter" />
                                <span className="af-filter-inline__title-text">Filtrer par</span>
                            </span>
                            <div className="af-filter-inline__field">
                                <label className="af-form__group-label" htmlFor="inputtext1">Nom du dataset</label>
                                <div className="af-form__text">
                                    <input className="af-form__input-text" id="inputtext1" name="inputtextname" onChange={onChangeFilter} type="text" />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <EmptyArrayManager items={items} emptyArrayMessage="Aucun élément">
                    <ItemsTable 
                        items={items}
                        filters={filters}
                        onChangePaging={onChangePaging}
                        onChangeFilter={onChangeFilter} />
                </EmptyArrayManager>
            </div>
        </>
    )
};

export default Home;
