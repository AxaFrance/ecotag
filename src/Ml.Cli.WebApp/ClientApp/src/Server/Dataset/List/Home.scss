@import '@axa-fr/react-toolkit-form-filter-inline/dist/filter-inline.scss';

.af-btn {
  &--quote {
    margin-bottom: 1em;
  }
  &--locked {
    background-color: green;
  }
  &--pending {
    background-color: orange;
  }
  &--locked-and-work-in-progress {
    background-color: darkgoldenrod;
  }
  &--none {
    background-color: #2a87d0;
  }
}

.row--datasets-filters {
  padding-bottom: 1.5rem;
}


.af-home__access-token{
  margin-left: 2em;
}