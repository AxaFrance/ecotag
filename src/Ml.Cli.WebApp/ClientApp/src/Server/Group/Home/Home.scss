@import '@axa-fr/react-toolkit-form-filter-inline/dist/filter-inline.scss';

.af-btn {
  &--quote {
    margin-bottom: 1em;
  }
}

.row--projects-filters{
  padding-bottom: 2rem;
}