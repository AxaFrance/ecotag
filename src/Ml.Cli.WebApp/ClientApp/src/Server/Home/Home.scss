@import '@axa-fr/react-toolkit-core/src/common/scss/core.scss';

.home {
  &__links-container {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
  }
  &__link-container {
    border-radius: 20px;
    width: 200px;
    height: 200px;
    background-color: #00008f;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.4em;
    text-align: center;
    transition: 0.3s;
    &--projects {
      background-color: $color-sante;
    }
    &--datasets {
      background-color: $color-epargne;
    }
    &--groups {
      background-color: $color-habitation;
    }
    &:hover {
      opacity: 0.8;
    }
  }
  &__link {
    color: #FFFFFF;
    font-weight: 400;
    font-size: 18px;
    margin-right: 20px;
    margin-bottom: 20px;
    &:hover {
      color: #FFFFFF;
    }
  }
}
