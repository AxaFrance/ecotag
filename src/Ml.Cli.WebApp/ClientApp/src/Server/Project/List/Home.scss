@import '@axa-fr/react-toolkit-form-filter-inline/dist/filter-inline.scss';

.af-btn {
  &--quote {
    margin-bottom: 1em;
  }
}

.row--projects-filters{
  padding-bottom: 2rem;
}

.projects-af-popover__wrapper > .af-popover__wrapper {
  width: 100%;
}

.af-popover__container--project-home{
  width: 100%;
}

.projects-af-popover__wrapper .af-popover__container .af-popover__arrow::before{
  background: #939393;
}

.projects-af-popover__wrapper .af-popover__container-pop {
  background: #939393;
}

.projects-af-popover__wrapper h3{
  color: white;
}