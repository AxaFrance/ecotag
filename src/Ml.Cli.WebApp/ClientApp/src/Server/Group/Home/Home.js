import React from 'react';
import PropTypes from 'prop-types';
import Title from '../../../TitleBar';
import New from './New';

import './Home.scss';
import '../../shared/Modal/modal.scss';
import EmptyArrayManager from "../../../EmptyArrayManager";
import ItemsTable from "./ItemsTable";

const Home = ({
  items,
  numberItemsTotal,
  filters,
  loaderMode,
  onChangePaging,
  onDeleteGroup,
  fields,
  isSubmitable,
  hasSubmit,
  onChangeCreateGroup,
  onSubmitCreateGroup,
  onUpdateUser,
}) => {
  return (
    <>
      <Title subtitle="Gestion des groupes d'annotateurs" title="Page Groupes"  />
      <div className="af-home container">
        <h1 className="af-title--content">Création de groupe</h1>
        <New
          disabled={isSubmitable}
          fields={fields}
          hasSubmit={hasSubmit}
          onChangeCreateGroup={onChangeCreateGroup}
          onSubmitCreateGroup={onSubmitCreateGroup}
        />
        <h1 className="af-title--content">
          {`Affichage des groupes - il y a actuellement (${numberItemsTotal}) groupes`}
        </h1>
        <EmptyArrayManager items={items} emptyArrayMessage="Aucun élément !">
          <ItemsTable
              items={items}
              filters={filters}
              loaderMode={loaderMode}
              onChangePaging={onChangePaging}
              onDeleteGroup={onDeleteGroup}
              onUpdateUser={onUpdateUser}
          />
        </EmptyArrayManager>
      </div>
    </>
  );
};

Home.defaultProps = {
  items: [],
  loaderMode: '',
};
Home.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      users: PropTypes.array,
    })
  ),
  fields: PropTypes.object,
  loaderMode: PropTypes.string,
  isSubmitable: PropTypes.bool,
  hasSubmit: PropTypes.bool,
  onChangeCreateGroup: PropTypes.func,
  onSubmitCreateGroup: PropTypes.func,
  onChangePaging: PropTypes.func,
  onDeleteGroup: PropTypes.func,
  onUpdateUser: PropTypes.func,
};

export default Home;
