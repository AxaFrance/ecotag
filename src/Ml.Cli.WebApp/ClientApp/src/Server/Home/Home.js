import React from "react";
import Title from "../../TitleBar";
import {Link} from 'react-router-dom';
import './Home.scss';
import withAuthentication, { Administateur, Annotateur, DataScientist } from "../withAuthentication";

export const Home = ({ user: { roles = [] } }) => (
    <div className="home">
        <Title title="Accueil" goButton={false}/>
        <div className="home__links-container">
            { roles.includes(Annotateur) ? <Link className="home__link" to="/projects">
                <div className="home__link-container home__link-container--projects">Projets</div>
            </Link> : null}
            { roles.includes(DataScientist) ? <Link className="home__link" to="/datasets">
                <div className="home__link-container home__link-container--datasets">Datasets</div>
            </Link>: null}
            { roles.includes(Administateur) ? <Link className="home__link" to="/groups">
                <div className="home__link-container home__link-container--groups">Groupes</div>
            </Link> : null}
        </div>
    </div>
);

export default withAuthentication()(Home);
