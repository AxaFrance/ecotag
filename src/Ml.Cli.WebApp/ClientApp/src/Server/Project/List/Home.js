import React from 'react';
import { Link,  } from 'react-router-dom';
import Title from '../../../TitleBar';
import './Home.scss';
import '../../shared/Modal/modal.scss';
import EmptyArrayManager from "../../../EmptyArrayManager";
import ItemsTable from "./ItemsTable";

const Home = ({ items, filters, onChangePaging, onChangeFilter, onChangeSort, onDeleteProject }) => {
  const numberItemsTotal = items && items.length ? items.length : 0;
  return (
      <>
        <Title title="Page projets" subtitle="tagger un ensemble de donnée" />
        <div className="af-home container">
          <Link className="btn af-btn af-btn--quote" to="/projects/new">
            <span className="af-btn__text">Nouveau projet</span>
          </Link>
          <h1 className="af-title--content">{`Vos projets en cours (${numberItemsTotal})`}</h1>
          <div className="row row--projects-filters">
            <div className="col">
              <div className="af-filter-inline">
              <span className="af-filter-inline__title">
                <span className="glyphicon glyphicon-filter" />
                <span className="af-filter-inline__title-text">Filtrer par</span>
              </span>
                <div className="af-filter-inline__field">
                  <label className="af-form__group-label" htmlFor="inputTextFilterProjects">
                    Nom du projet
                  </label>
                  <div className="af-form__text">
                    <input
                        className="af-form__input-text"
                        id="inputTextFilterProjects"
                        name="inputTextFilterProjects"
                        onChange={event => onChangeFilter(event.target.value)}
                        type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <EmptyArrayManager items={items} emptyArrayMessage="Aucun élément">
            <ItemsTable
                items={items}
                filters={filters}
                onChangePaging={onChangePaging}
                onChangeSort={onChangeSort}
                onDeleteProject={onDeleteProject}
            />
          </EmptyArrayManager>
        </div>
      </>
  );
};

export default Home;
