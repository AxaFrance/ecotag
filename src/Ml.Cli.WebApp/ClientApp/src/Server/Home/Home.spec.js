import React from "react";
import '@testing-library/jest-dom';
import {render} from '@testing-library/react';
import { Home } from "./Home";
import {BrowserRouter as Router} from "react-router-dom";

describe('Check page shifting', () => {
    test('Render home page and go to compare page', async () => {
        const { asFragment  } = render(<Router basename="/"><Home /></Router>);

        expect(asFragment()).toMatchSnapshot();
    });
});
