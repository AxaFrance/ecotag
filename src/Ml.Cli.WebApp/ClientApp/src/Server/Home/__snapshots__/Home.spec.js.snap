// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home "" Render home page 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    />
    <p>
      Vous n'avez aucun rôle attribué à votre profile.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Home "ECOTAG_ANNOTATEUR" Render home page 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/projects"
      >
        <div
          class="home__link-container home__link-container--projects"
        >
          Projets
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home "ECOTAG_DATA_SCIENTIST,ECOTAG_ANNOTATEUR" Render home page 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/projects"
      >
        <div
          class="home__link-container home__link-container--projects"
        >
          Projets
        </div>
      </a>
      <a
        class="home__link"
        href="/datasets"
      >
        <div
          class="home__link-container home__link-container--datasets"
        >
          Datasets
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home "ECOTAG_DATA_SCIENTIST,ECOTAG_ANNOTATEUR,ECOTAG_ADMINISTRATEUR" Render home page 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/projects"
      >
        <div
          class="home__link-container home__link-container--projects"
        >
          Projets
        </div>
      </a>
      <a
        class="home__link"
        href="/datasets"
      >
        <div
          class="home__link-container home__link-container--datasets"
        >
          Datasets
        </div>
      </a>
      <a
        class="home__link"
        href="/groups"
      >
        <div
          class="home__link-container home__link-container--groups"
        >
          Equipe
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
